File: java_api/src/main/java/utils/ConfigurationLoader.java
package utils;

import org.json.JSONObject;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.io.*;

public class ConfigurationLoader {
  Logger mLogger = LoggerFactory.getLogger(ConfigurationLoader.class);

  private static ConfigurationLoader mInstance = null;

  private String mDBName = "";
  private String mDBHost = "";
  private String mDBUser = "";
  private String mDBPass = "";
  private String mAWSKey = "";
  private String mAWSUser = "";
  private String mAWSInstance = "";

  public static ConfigurationLoader getInstance() {
    if(mInstance == null) {
      try {
        ConfigurationLoader conf = new ConfigurationLoader();
        mInstance = conf.load();
      }
      catch(Exception e){
        e.printStackTrace();
      }
    }

    return mInstance;
  }

  private ConfigurationLoader load() {
    String path = System.getProperty("ConfigPath");

    if (path == null)
    {
      mLogger.error("Invalid configuration path");
      System.exit(1);
    }

    try {
      File file = new File(System.getProperty("ConfigPath"));
      BufferedReader br = new BufferedReader(new FileReader(file));

      String st;
      StringBuffer sb = new StringBuffer();
      while ((st = br.readLine()) != null)
        sb.append(st);

      JSONObject obj = new JSONObject(sb.toString());

      this.mDBName = obj.getString("db_name");
      this.mDBHost = obj.getString("db_host");
      this.mDBUser = obj.getString("db_user");
      this.mDBPass = obj.getString("db_password");
      this.mAWSKey = obj.getString("aws_key");
      this.mAWSUser = obj.getString("aws_user");
      this.mAWSInstance = obj.getString("aws_instance");
    }
    catch(IOException ex) {
      mLogger.error("Error reading configuration file");
      return null;
    }

    return this;
  }


  public String getDBName() {
    return this.mDBName;
  }

  public String getDBHost() {
    return this.mDBHost;
  }

  public String getDBUser() {
    return this.mDBUser;
  }

  public String getDBPass() {
    return this.mDBPass;
  }

  public String getAWSKey() {
    return this.mAWSKey;
  }

  public String getAWSUser() {
    return this.mAWSUser;
  }

  public String getAWSInstance() {
    return this.mAWSInstance;
  }

}

File: kotlin_api/src/main/kotlin/utils/ConfigurationLoader.kt
package utils

import org.json.JSONObject
import org.slf4j.LoggerFactory
import java.io.File
import java.io.IOException

class ConfigurationLoader {
    private val mLogger = LoggerFactory.getLogger(ConfigurationLoader::class.java)

    var dbName = ""
        private set
    var dbHost = ""
        private set
    var dbUser = ""
        private set
    var dbPass = ""
        private set
    var awsKey = ""
        private set
    var awsUser = ""
        private set
    var awsInstance = ""
        private set

    @Throws(Exception::class)
    private fun load(): ConfigurationLoader? {
        val path = System.getProperty("ConfigPath")

        if (path == null) {
            mLogger.error("Invalid configuration path")
            System.exit(1)
        }

        val file = File(path)
        try {
            val content = file.readText()

            val obj = JSONObject(content)

            this.dbName = obj.getString("db_name")
            this.dbHost = obj.getString("db_host")
            this.dbUser = obj.getString("db_user")
            this.dbPass = obj.getString("db_password")
            this.awsKey = obj.getString("aws_key")
            this.awsUser = obj.getString("aws_user")
            this.awsInstance = obj.getString("aws_instance")
        }
        catch(ex: IOException){
            mLogger.error("The configuration file was not found")
            return null
        }


        return this
    }

    companion object {
        private var mInstance: ConfigurationLoader? = null

        val instance: ConfigurationLoader?
            get() {
                if (mInstance == null) {
                    try {
                        val conf = ConfigurationLoader()
                        mInstance = conf.load()
                    } catch (e: Exception) {
                        LoggerFactory.getLogger(ConfigurationLoader::class.java)
                            .error("The configuration file was not found")
                    }

                }

                return mInstance
            }
    }

}
